Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -202,53 +161,34 @@ open class BpkText @JvmOverloads constructor(
   }
 }
 
-private fun internalGetFontFromTheme(context: Context, weight: BpkText.Weight = BpkText.Weight.NORMAL): Typeface? {
-  val a = context.theme.obtainStyledAttributes(null, R.styleable.BpkText, R.attr.bpkTextStyle, 0)
-
-  val nullIfInvalid = { resId: Int -> if (resId == -1) null else resId }
-
-  val fontBase = a.getResourceId(R.styleable.BpkText_fontFamilyBase, -1).let(nullIfInvalid)
-  val fontEmphasized = a.getResourceId(R.styleable.BpkText_fontFamilyEmphasized, -1).let(nullIfInvalid)
-  val fontHeavy = a.getResourceId(R.styleable.BpkText_fontFamilyHeavy, -1).let(nullIfInvalid)
-
-  val fontResource = when (weight) {
-    BpkText.Weight.EMPHASIZED -> fontEmphasized
-    BpkText.Weight.HEAVY -> fontHeavy
-    BpkText.Weight.NORMAL -> fontBase
-  }
-
-  a.recycle()
-
-  return fontResource?.let { FontCache[it, context] }
-}
-
 private fun internalGetFont(context: Context, textStyle: Int = BpkText.BASE, weight: BpkText.Weight = BpkText.Weight.NORMAL): BpkText.FontDefinition {
-  val styleRes = getStyleId(textStyle, weight)
+  val styleRes = getStyleId(context, textStyle, weight)
 
   val textStyleAttributes = context.obtainStyledAttributes(styleRes, R.styleable.BpkTextStyle)
-  val fontFamily = textStyleAttributes.getString(R.styleable.BpkTextStyle_android_fontFamily) ?: "sans-serif"
   val fontSize = textStyleAttributes.getDimensionPixelSize(R.styleable.BpkTextStyle_android_textSize, ResourcesUtil.dpToPx(16, context))
   val letterSpacing = textStyleAttributes.getFloat(R.styleable.BpkTextStyle_android_letterSpacing, -1f)
     .let { if (it == -1f) null else it }
 
   textStyleAttributes.recycle()
 
-  val fontFromTheme = internalGetFontFromTheme(context, weight)
-
-  if (fontFromTheme != null) {
-    return BpkText.FontDefinition(fontFromTheme, fontSize, null, isCustomFont = true)
-  }
-
-  return BpkText.FontDefinition(Typeface.create(fontFamily, Typeface.NORMAL), fontSize, letterSpacing)
+  return FontFamilyResolver(context).getForWeight(weight)?.let {
```

Comment:
What will allocate a new object every time the BpkText is used. We need to make it static.